lib/jobs.rb

```diff
@@ -27,8 +27,15 @@ def self.establish_connection!
     end
   end
 
-  def self.search(options)
-    @usajobs_api_connection.get(@endpoint, options).body if query_eligible?(options[:Keyword])
+  def self.job_scrub(keyword)
+    scrubbed_keyword = keyword.gsub(/#{JOB_RELATED_KEYWORDS}/, '').strip
```

Comment:
I think you want to use `squish` instead of `strip` here, to remove extra inner spaces too.

Reply:
Also, you could use `remove` here instead of `gsub` (either way is fine w/ me, although I find `remove` reads a little better):
```ruby
keyword.remove(/#{JOB_RELATED_KEYWORDS}/).squish
```

Reply:
Your regex needs word boundaries so that we don't partially strip words. Examples:
```ruby
> "grand reopening".gsub(/#{JOB_RELATED_KEYWORDS}/, '')
"grand re"

> "seasonality".gsub(/#{JOB_RELATED_KEYWORDS}/, '')
"ity"
```

```

Reply:
@peggles2 , the regex still needs work:
```ruby
> Jobs.scrub_keyword('grand re')
"grand re"
```

We should also have test cases for these examples.

Reply:
These are also not working properly:
```ruby
>Jobs.scrub_keyword('seasonal jobs')
"seasonal jobs"

> Jobs.scrub_keyword('job posting')
"job posting"
```

Reply:
@peggles2 , I should have thought of this earlier, but I ran your scrubber against the list of queries from prod. Github didn't like the file, so I've attached it to the JIRA issue: https://cm-jira.usa.gov/secure/attachment/68014/scrubbed_job_queries.csv

The list shows the original query and the scrubbed version. It's looking pretty good, but there are some cases that jump out that we should cover (and add test cases for):
- the scrubber should be case-insensitive, so that it will scrub out "Jobs", etc.
- we should scrub out the generic Spanish terms, such as "empleo", "trabajo", etc.
- we should scrub out `(opportunit|vacanc)(y|ies)`

With those tweaks, this should be good to go.

Reply:
If we do that we should just scrub all the equivalent spanish words that matches our generic job terms. which then leaves us basically scrubbing everything but internship, seasonal, federal, and gobierno.  I ran some tests and found a bug in the current code.

  JOB_RELATED_KEYWORDS = '((position|opening|posting|job|employment|intern(ship)?|seasonal|trabajo|puesto|empleo|vacante)s?|(opportunit|vacanc)(y|ies))|(posicion|ocupacion|oportunidad|federal)es|gobierno'

Place that in Rubular and you will see that posicion, ocupacion, oportunidad and federal will not match unless you add "es".  I believe thats to make it plural?

The correct way is
  JOB_RELATED_KEYWORDS = '((position|opening|posting|job|employment|intern(ship)?|seasonal|trabajo|puesto|empleo|vacante)s?|(opportunit|vacanc)(y|ies))|(posicion|ocupacion|oportunidad|federal)(es\b|\b)|gobierno'